common/lib/dependabot/update_checkers/base.rb

```diff
@@ -8,12 +8,11 @@ module Dependabot
   module UpdateCheckers
     class Base
       attr_reader :dependency, :dependency_files, :repo_contents_path,
-                  :credentials, :ignored_versions, :raise_on_ignored,
-                  :security_advisories, :requirements_update_strategy,
-                  :options
+                  :credentials, :raise_on_ignored, :security_advisories,
+                  :requirements_update_strategy, :options
 
       def initialize(dependency:, dependency_files:, repo_contents_path: nil,
-                     credentials:, ignored_versions: [],
+                     credentials:, ignore_conditions: [], ignored_versions: [],
```

Comment:
Concern: accepting both `ignore_conditions` and `ignored_versions` here is a weird smell.  Should `ignored_versions` be considered deprecated?

Since `ignore_conditions` is an array of items that is filtered by `dependency`, I don't get the value of including the entire list (including entries not relevant to `dependency` of interest).